llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskRunnerCallable.java

```diff
@@ -226,10 +226,6 @@ protected TaskRunner2Result callInternal() throws Exception {
 
       // TODO Consolidate this code with TezChild.
       runtimeWatch.start();
-      if (fsTaskUgi == null) {
```

Comment:
should we at least add `Preconditions.checkArgument(fsTaskUgi != null)`?

Reply:
we can, however, it won't help if it's null, but at least I can add a custom message for user convenience's sake

Reply:
could we add this precondition check to the TaskRunnerCallable constructor?

Reply:
ack